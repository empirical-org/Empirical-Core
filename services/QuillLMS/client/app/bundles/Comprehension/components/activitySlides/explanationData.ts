export const explanationData = {
  1: {
    buttonText: "Next",
    header: "This activity is not graded",
    imageData: {
      imageAlt: "An illustration of an A+ that is crossed out",
      imageUrl: "images/comprehension/no-grades.svg"
    },
    isBeta: false,
    step: 2,
    subtext: "This is a safe space to practice your writing, so it won't be graded. Your teacher will see your revisions, but there are no scores or points."
  },
  2: {
    buttonText: "Next",
    header: "We use a feedback bot",
    imageData: {
      imageAlt: "An illustration of a smiling robot head",
      imageUrl: "images/comprehension/feedback-bot.svg"
    },
    isBeta: true,
    step: 3,
    subtext: "We use artificial intelligence (AI) to help us give you feedback on your writing. You should know that AI isn't always correct."
  },
  3: {
    buttonText: "Start",
    header: "All writers revise",
    imageData: {
      imageAlt: "An illustration of a pencil circled with a revision",
      imageUrl: "images/comprehension/revising-pencil.svg"
    },
    isBeta: false,
    step: 4,
    subtext: "You'll be able to revise each sentence up to five times. We give you feedback because we want to help you write a stronger sentence."
  }
}
