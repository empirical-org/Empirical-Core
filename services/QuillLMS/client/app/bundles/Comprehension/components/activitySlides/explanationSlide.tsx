import * as React from "react";

import Footer from './footer';

export const ExplanationSlide = ({ slideData, onHandleClick }) => {
  const { buttonText, header, imageData, isBeta, step, subtext } = slideData;
  const { imageAlt, imageUrl } = imageData;

  return(
    <div className="explanation-slide-container">
      <section id="information-section">
        <p className="subtext">Good to know</p>
        <section id="header-container">
          <p id="header">{header}</p>
          {isBeta && <div id="beta-tag">BETA</div>}
        </section>
        <img alt={imageAlt} id="image" src={`${process.env.CDN_URL}/${imageUrl}`} />
        <p className="subtext">{subtext}</p>
      </section>
      <Footer buttonText={buttonText} onHandleClick={onHandleClick} step={step} />
    </div>
  );
}

export default ExplanationSlide;
