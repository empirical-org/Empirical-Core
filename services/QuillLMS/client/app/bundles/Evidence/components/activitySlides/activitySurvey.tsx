import * as React from 'react';

import useFocus from '../../../Shared/hooks/useFocus';
import { TrackAnalyticsEvent } from "../../actions/analytics";
import { Events } from '../../modules/analytics';

const baseImgSrc = `${process.env.CDN_URL}/images/pages/evidence`
const checkMarkSrc = `${baseImgSrc}/components-selection-controls-dark-enabled-selected.svg`

const SURVEY_QUESTION = "What did you think of this activity?"

const emojis = [
  {
    alt: 'Angry face',
    src: `${baseImgSrc}/angry-face.svg`
  },
  {
    alt: 'Frowning face',
    src: `${baseImgSrc}/frowning-face.svg`
  },
  {
    alt: 'Neutral face',
    src: `${baseImgSrc}/neutral-face.svg`
  },
  {
    alt: 'Smiling face',
    src: `${baseImgSrc}/smiling-face.svg`
  },
  {
    alt: 'Excited face',
    src: `${baseImgSrc}/excited-face.svg`
  }
]

const positiveMultipleChoiceOptions = [
  'I thought the feedback was helpful',
  'I thought completing the sentences was a good challenge',
  'I thought completing the sentences was easy',
  'I had no trouble reading the text',
  'I thought this topic was interesting'
]

const negativeMultipleChoiceOptions = [
  'I got stuck on the feedback',
  'I thought completing the sentences was too difficult',
  'I thought completing the sentences was too easy',
  'I thought the text was too difficult',
  'I thought this topic was boring'
]

const EmojiButton = ({ src, alt, emojiNumber, setSelectedEmoji, selectedEmoji, }) => {
  function handleClick() { setSelectedEmoji(emojiNumber) }

  return (
    <button className={`emoji-button ${selectedEmoji === emojiNumber ? 'selected' : ''}`} onClick={handleClick} type="button">
      <img alt={alt} src={src} />
    </button>
  )
}

const MultipleChoiceOption = ({ text, selectedMultipleChoiceOptions, setSelectedMultipleChoiceOptions, }) => {
  const optionIsSelected = selectedMultipleChoiceOptions.find(opt => opt === text)

  function handleCheckboxClick() {
    const newSelectedMultipleChoiceOptions = optionIsSelected ? selectedMultipleChoiceOptions.filter(opt => opt !== text) : selectedMultipleChoiceOptions.concat(text)
    setSelectedMultipleChoiceOptions(newSelectedMultipleChoiceOptions)
  }

  let checkbox = <span aria-label="Unchecked checkbox" className="quill-checkbox unselected" />
  if (optionIsSelected) {
    checkbox = (<span aria-label="Checked checkbox" className="quill-checkbox selected" >
      <img alt="Checked checkbox" src={checkMarkSrc} />
    </span>)
  }

  return (
    <button className={`multiple-choice-option ${optionIsSelected ? 'selected' : ''}`} onClick={handleCheckboxClick} type="button">
      {checkbox}
      <span className="text-container">{text}</span>
    </button>
  )
}

const ActivitySurvey = ({ activity, dispatch, sessionID, saveActivitySurveyResponse, setSubmittedActivitySurvey, }) => {
  const [selectedEmoji, setSelectedEmoji] = React.useState(null)
  const [selectedMultipleChoiceOptions, setSelectedMultipleChoiceOptions] = React.useState([])

  const [containerRef, setContainerFocus] = useFocus()

  React.useEffect(() => {
    setContainerFocus()
  }, [])

  React.useEffect(() => { setSelectedMultipleChoiceOptions([]) }, [selectedEmoji])

  function mapMultipleChoiceOptionsForEventParams() {
    const options = {}
    positiveMultipleChoiceOptions.forEach(option => {
      const key = option.split(' ').join('_')
      options[key] = false
    })
    negativeMultipleChoiceOptions.forEach(option => {
      const key = option.split(' ').join('_')
      options[key] = false
    })
    selectedMultipleChoiceOptions.forEach(option => {
      const key = option.split(' ').join('_')
      options[key] = true
    });
    return options
  }

  function handleSend() {
    if (!selectedMultipleChoiceOptions.length) { return }
    const activitySurveyResponse = {
      emoji_selection: selectedEmoji,
      multiple_choice_selections: selectedMultipleChoiceOptions,
      survey_question: SURVEY_QUESTION,
    }
    const trackingParams = {
      activity_name: activity?.title,
      tool_name: "Reading",
      rating: selectedEmoji,
      ...mapMultipleChoiceOptionsForEventParams()
    }
    const callback = () => setSubmittedActivitySurvey(true)
    dispatch(TrackAnalyticsEvent(Events.STUDENT_RATED_AN_ACTIVITY, trackingParams))
    saveActivitySurveyResponse({ sessionID, activitySurveyResponse, callback, })
  }

  function handleLinkClick() { window.location.href = process.env.DEFAULT_URL }

  let activitySurveyHeader
  let multipleChoiceSection

  if (selectedEmoji) {
    const multipleChoiceOptions = selectedEmoji > 3 ? positiveMultipleChoiceOptions : negativeMultipleChoiceOptions
    const multipleChoiceElements = multipleChoiceOptions.map(opt => <MultipleChoiceOption key={opt} selectedMultipleChoiceOptions={selectedMultipleChoiceOptions} setSelectedMultipleChoiceOptions={setSelectedMultipleChoiceOptions} text={opt} />)
    multipleChoiceSection = (<section>
      <h1>Select one or more options:</h1>
      {multipleChoiceElements}
    </section>)
  } else {
    activitySurveyHeader = (<section>
      <p>We need your help</p>
      <h1>{SURVEY_QUESTION}</h1>
    </section>)
  }

  const emojiButtons = emojis.map((emoji, i) => (
    <EmojiButton
      alt={emoji.alt}
      emojiNumber={i + 1}
      key={emoji.alt}
      selectedEmoji={selectedEmoji}
      setSelectedEmoji={setSelectedEmoji}
      src={emoji.src}
    />)
  )

  const sendButtonClassName = selectedMultipleChoiceOptions.length ? 'quill-button large secondary outlined focus-on-dark' : 'quill-button large disabled contained focus-on-dark'

  return (
    <div className="activity-follow-up-container activity-survey-container no-focus-outline" ref={containerRef} tabIndex={-1}>
      <div className="activity-survey-content">
        {activitySurveyHeader}
        <div className="emoji-buttons">{emojiButtons}</div>
        {multipleChoiceSection}
      </div>
      <div className="button-section">
        <a className="focus-on-dark" href={process.env.DEFAULT_URL} onClick={handleLinkClick}>Skip</a>
        <button className={sendButtonClassName} onClick={handleSend} type="button">Send</button>
      </div>
    </div>
  )
}

export default ActivitySurvey
