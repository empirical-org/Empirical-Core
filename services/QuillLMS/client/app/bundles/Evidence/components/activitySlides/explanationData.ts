export const explanationData = {
  1: {
    buttonText: "Next",
    header: "All writers revise",
    imageData: {
      imageAlt: "An illustration of a pencil circled with a revision",
      imageUrl: "images/evidence/revising-pencil.svg"
    },
    isBeta: false,
    step: 2,
    subtext: "Your goal is to practice and improve your writing skills.<br /><br />We will provide you with a lot of feedback on your writing. Youâ€™ll be able to revise each sentence up to five times so that you can become a stronger writer."
  },
  2: {
    buttonText: "Next",
    header: "We use a feedback bot",
    imageData: {
      imageAlt: "An illustration of a smiling robot head",
      imageUrl: "images/evidence/feedback-bot.svg"
    },
    isBeta: true,
    step: 3,
    subtext: "We use artificial intelligence (AI) to help us give you feedback on your writing. You should know that AI isn't always correct."
  },
  3: {
    buttonText: "Start",
    header: "Your teacher will see your score and writing",
    imageData: {
      imageAlt: "An illustration of a report page with a magnifying glass over it",
      imageUrl: "images/evidence/evidence-report.svg"
    },
    isBeta: false,
    step: 4,
    subtext: "Once the activity is complete, both you and your teacher will see a report with your score and your responses for each prompt."
  },
}
