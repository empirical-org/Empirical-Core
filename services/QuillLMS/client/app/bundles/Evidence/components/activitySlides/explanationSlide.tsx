import * as React from "react";

import Footer from './footer';

import useFocus from '../../../Shared/hooks/useFocus';

export const ExplanationSlide = ({ slideData, onHandleClick }) => {
  const { buttonText, header, imageData, isBeta, step, subtext } = slideData;
  const { imageAlt, imageUrl } = imageData;

  const [containerRef, setContainerFocus] = useFocus()

  React.useEffect(() => {
    setContainerFocus()
  }, [slideData])

  return(
    <div className="explanation-slide-container">
      <section className="no-focus-outline" id="information-section" ref={containerRef} tabIndex={-1}>
        <p aria-hidden={true} className="subtext">Good to know</p>
        <section id="header-container">
          <h1>{header}</h1>
          {isBeta && <div id="beta-tag">BETA</div>}
        </section>
        <img alt={imageAlt} id="image" src={`${import.meta.env.VITE_PROCESS_ENV_CDN_URL}/${imageUrl}`} />
        <p className="subtext">{subtext}</p>
      </section>
      <Footer buttonText={buttonText} onHandleClick={onHandleClick} step={step} />
    </div>
  );
}

export default ExplanationSlide;
