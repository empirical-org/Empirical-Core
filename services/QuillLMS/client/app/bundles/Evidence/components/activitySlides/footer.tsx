import * as React from "react";

export const Footer = ({ buttonText, onHandleClick, step }) => {

  function handleClick() {
    onHandleClick();
  }

  return(
    <section id="button-container">
      <span />
      <button className="quill-button large secondary outlined focus-on-dark" onClick={handleClick} type="submit">{buttonText}</button>
      <div id="step-indictator-container">
        <div className={`step-indicator ${step === 1 ? 'active' : ''}`} />
        <div className={`step-indicator ${step === 2 ? 'active' : ''}`} />
        <div className={`step-indicator ${step === 3 ? 'active' : ''}`} />
        <div className={`step-indicator ${step === 4 ? 'active' : ''}`} />
      </div>
    </section>
  );
}

export default Footer;
