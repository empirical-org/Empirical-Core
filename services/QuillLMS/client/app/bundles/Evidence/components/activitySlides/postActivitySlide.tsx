import * as React from "react";

import { BECAUSE, BUT, SO } from '../../../Shared/index'
import { PromptInterface } from "../../../Staff/interfaces/evidenceInterfaces";

const FIRST_STRONG_EXAMPLE = 'first_strong_example';
const SECOND_STRONG_EXAMPLE = 'second_strong_example';
const conjunctions = [BECAUSE, BUT, SO];

interface PostActivitySlideProps {
  handleClick: () => void,
  prompts: PromptInterface[],
  responses: any
}

export const PostActivitySlide = ({ handleClick, prompts, responses }: PostActivitySlideProps) => {

  function getStrongExemplar(conjunction: string, property: string) {
    const prompt = prompts.filter(prompt => prompt.conjunction === conjunction)[0];
    if (!prompt) { return }
    const stemWithoutConjunction = prompt.text.replace(conjunction, '');
    const exemplar = prompt[property];
    return <p>{stemWithoutConjunction}<u>{conjunction}</u>{exemplar}</p>;
  }

  function getResponsesForConjunction(conjunction: string) {
    const responsesKeyHash = {}
    const responsesKeys = Object.keys(responses);
    conjunctions.forEach((conjunctionString, i) => responsesKeyHash[conjunctionString] = responsesKeys[i]);
    return responses[responsesKeyHash[conjunction]];
  }

  function getResponse(conjunction: string) {
    const responses = getResponsesForConjunction(conjunction)
    const lastResponseText = responses[responses.length - 1].entry;
    const splitResponse = lastResponseText.split(conjunction);
    return <p>{splitResponse[0]}<u>{conjunction}</u>{splitResponse[1]}</p>
  }

  function getRevisionCopy() {
    let count = 0;
    const noRevisionsCopy = 'You completed the activity with 0 revisions!';
    if(!responses) { return noRevisionsCopy }
    Object.keys(responses).map(key => {
      count += responses[key].length;
    });
    if(count <= 3) { return noRevisionsCopy }
    // we subtract 3 to account for a student having 3 optimal first attempts
    const calculatedCount = count - 3;
    if(calculatedCount === 1) {
      return 'You completed the activity with 1 revision!';
    }
    return `You completed the activity with ${calculatedCount} revisions!`;
  }

  function renderResponseAndExamplarsSection(conjunction: string) {
    return(
      <section className="response-exemplars-section">
        <section className="response-section">
          <div className="response-exemplar-header">
            <p className="sub-header-text">You wrote</p>
          </div>
          <section className="response-box">{getResponse(conjunction)}</section>
        </section>
        <section className="exemplars-section">
          <div className="response-exemplar-header">
            <p className="sub-header-text">Example Responses</p>
          </div>
          <section className="exemplar-box">{getStrongExemplar(conjunction, FIRST_STRONG_EXAMPLE)}</section>
          <section className="exemplar-box">{getStrongExemplar(conjunction, SECOND_STRONG_EXAMPLE)}</section>
        </section>
      </section>
    );
  }

  return(
    <div className="post-activity-slide-container">
      <section id="information-section">
        <img alt="An illustration of a party popper" id="celebration-vector" src={`${process.env.CDN_URL}/images/evidence/party-celebration.svg`} />
        <p id="revision-text">{getRevisionCopy()}</p>
        <p className="slide-sub-text" id="second-sub-text">Revising is a sign of being a thoughtful writer. Be proud of the work you did today, and celebrate your success!</p>
        <section id="reminder-badge-section">
          <img alt="An illustration of an A+ that is crossed out" id="grade-badge" src={`${process.env.CDN_URL}/images/evidence/paper-check.svg`} />
          <section id="reminder-text-section">
            <p className="sub-header-text">Reminder about grades</p>
            <p className="sub-header-subtext">Your teacher will see all of your revisions, but this activity was for practice, so it isnâ€™t graded.</p>
          </section>
        </section>
      </section>
      <section className="responses-exemplars-container">
        <section className="review-response-header-section">
          <p className="sub-header-text">Review the other responses.</p>
          <p className="sub-header-subtext">How is your response similar or different from these responses?</p>
        </section>
        {renderResponseAndExamplarsSection(BECAUSE)}
        {renderResponseAndExamplarsSection(BUT)}
        {renderResponseAndExamplarsSection(SO)}
      </section>
      <section id="button-container">
        <button className="quill-button large secondary outlined focus-on-dark" onClick={handleClick} type="button">Next</button>
      </section>
    </div>
  );
}

export default PostActivitySlide;
