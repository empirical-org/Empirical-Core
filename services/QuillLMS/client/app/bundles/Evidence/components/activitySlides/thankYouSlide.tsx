import * as React from 'react'

const applaudingSrc = `${process.env.CDN_URL}/images/pages/evidence/applauding.svg`

const ThankYouSlide = () => {
  const backToDashboard = () => {
    window.location.href = process.env.DEFAULT_URL
  }

  return (<div className="activity-follow-up-container thank-you-slide-container">
    <div className="thank-you-content">
      <section>
        <p>Your feedback was sent</p>
        <h1>Thanks for sharing your feedback! </h1>
      </section>
      <img alt="Two hands clapping together" src={applaudingSrc} />
    </div>
    <div className="button-section">
      <a className='quill-button large secondary outlined focus-on-dark' href={process.env.DEFAULT_URL} onClick={backToDashboard}>Back to my dashboard</a>
    </div>
  </div>)
}

export default ThankYouSlide
