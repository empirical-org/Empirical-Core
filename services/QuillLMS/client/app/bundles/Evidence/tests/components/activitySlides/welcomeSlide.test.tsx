import * as React from 'react';
import { shallow } from 'enzyme';

import { WelcomeSlide } from '../../../components/activitySlides/welcomeSlide';

describe('WelcomeSlide Component', () => {
  const mockProps = {
    onHandleClick: jest.fn()
  }
  let component = shallow(<WelcomeSlide {...mockProps} />);

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
