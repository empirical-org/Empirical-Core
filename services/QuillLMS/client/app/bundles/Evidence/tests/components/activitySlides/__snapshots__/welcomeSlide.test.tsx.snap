// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeSlide Component should match snapshot 1`] = `
<div
  className="explanation-slide-container"
  id="welcome-slide-container"
>
  <section
    id="information-section"
  >
    <p
      className="subtext"
    >
      Hi there!
    </p>
    <section
      id="header-container"
    >
      <p
        id="header"
      >
        Welcome to Quill Evidence
      </p>
    </section>
    <section
      id="instructions-container"
    >
      <div
        className="instruction-container"
      >
        <div
          className="step-number"
        >
          1
        </div>
        <p
          className="step-text"
        >
          Read and highlight text
        </p>
        <img
          alt="A sample article about volcanoes showing a sentence highlighted with the cursor over the highlighted sentence"
          className="step-image"
          src="undefined/images/comprehension/article-highlighted.svg"
        />
      </div>
      <div
        className="instruction-container"
      >
        <div
          className="step-number"
        >
          2
        </div>
        <p
          className="step-text"
        >
          Write sentences using what you read
        </p>
        <img
          alt="A sample question asking a student to fill in the rest of a sentence"
          className="step-image"
          src="undefined/images/comprehension/sentence-prompt.svg"
        />
      </div>
      <div
        className="instruction-container"
      >
        <div
          className="step-number"
        >
          3
        </div>
        <p
          className="step-text"
        >
          Revise based on feedback
        </p>
        <img
          alt="A sample feedback asking a student to add more detail to their sentence"
          className="step-image"
          src="undefined/images/comprehension/sentence-feedback.svg"
        />
      </div>
    </section>
  </section>
  <Component
    buttonText="Next"
    onHandleClick={[MockFunction]}
    step={1}
  />
</div>
`;
